src/tools/miri/src/machine.rs

```diff
@@ -976,6 +976,20 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for MiriMachine<'mir, 'tcx> {
         throw_machine_stop!(TerminationInfo::Abort(msg))
     }
 
+    fn unwind_terminate(ecx: &mut InterpCx<'mir, 'tcx, Self>) -> InterpResult<'tcx> {
+        // Call the lang item.
+        let panic = ecx.tcx.lang_items().panic_cannot_unwind().unwrap();
+        let panic = ty::Instance::mono(ecx.tcx.tcx, panic);
+        ecx.call_function(
+            panic,
+            Abi::Rust,
+            &[],
+            None,
+            StackPopCleanup::Goto { ret: None, unwind: mir::UnwindAction::Unreachable },
+        )?;
+        Ok(())
+    }
+
```

Comment:
I think it would be nice if this function returned `InterpResult<'tcx, !>` signifying that it would never return an `Ok`. Can we assume that this `call_function` will always return an `Err`, or does the error happen at a later stage?

Reply:
This cannot return `!`. This just pushes the stack frame, and the interpreter loop has more work to do until eventually `panic_cannot_unwind` calls `abort`.

Reply:
Ah okay. Thanks for clarifying!